private/bufpkg/bufcas/file_node.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufcas
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	storagev1beta1 "github.com/bufbuild/buf/private/gen/proto/go/buf/registry/storage/v1beta1"
+	"github.com/bufbuild/buf/private/pkg/normalpath"
+)
+
+// FileNode is a path and associated digest.
+type FileNode interface {
```

Comment:
This is intended to match the naming of the registry API - `Digest, Blob, FileNode` all have direct corollaries, and since we don't have the concept of `Manifest` in the registry API, I'd like to keep it like this.